### ISSUE TITLE
Element Length

### ISSUE THREAD

**Issue by amit-choudhary** [ganisivakumar/ruby_tutorial#7]
1. Spaces needed around operators.
2. Use proper method name.
3. Use proper names for variables.
4. Please try to include question in main.rb
5. lenght_finder should not be responsible for printing result.

**Comment by ganisivakumar** [comment]
I changed method name to group_by_length
Fixed style guide errors
I changed all variable names.

**Comment by amit-choudhary** [comment]
What is the difference between Hash.new([]) and Hash.new { |hash, key| hash[key] = [] } ?

**Comment by amit-choudhary** [comment]
Also..Are you referring some source to do these exercises?

**Comment by ganisivakumar** [comment]
In Hash.new([]) all values for all keys is changing to the same,but in later it is changing only for a particular key,value pair.

**Comment by ganisivakumar** [comment]
yes,I am referring some code links to optimise my initial code.You can see the difference from old_files folder for some particular changes

**Comment by amit-choudhary** [comment]
1. What is the difference between Hash.new { |hash, key| hash[key] = [] } and Hash.new { |hash, key| hash[key] = Array.new } ?
2. What is the difference between Hash.new { |hash, key| hash[key] = 0 } and Hash.new(0) ?

**Comment by ganisivakumar** [comment]
1)Both are identical.If there is any argument to pass it is better to use Array.new with some argument.
2)Each function has their own limitations.In first approach,key is set to the '0' but we need to access the key before, to work in required way.In second approach,every key is not set to the '0',so it will fail if we access the key.

**Comment by amit-choudhary** [comment]
Did you try second one?

**Comment by ganisivakumar** [comment]
Yes,I used it in count_alphabets exercise,where I am overiding default value and printing.
